src/ads/MobileAd.ts

```diff
@@ -16,26 +16,33 @@
  */
 
 import { EmitterSubscription } from 'react-native';
+import { isFunction, isOneOf } from '../common';
 import { NativeError } from '../internal/NativeError';
-import { RewardedAdEventType } from '../RewardedAdEventType';
 import { AdEventType } from '../AdEventType';
-import { AdEventListener } from '../types/AdEventListener';
+import { RewardedAdEventType } from '../RewardedAdEventType';
+import { AdEventListener, AdEventPayload } from '../types/AdEventListener';
+import { AdEventsListener } from '../types/AdEventsListener';
 import { RequestOptions } from '../types/RequestOptions';
 import { MobileAdsModuleInterface } from '../types/MobileAdsModule.interface';
 import { RewardedAdReward } from '../types/RewardedAdReward';
 
+type EventType = AdEventType | RewardedAdEventType;
+
 export class MobileAd {
-  _type: 'app_open' | 'interstitial' | 'rewarded';
-  _googleMobileAds: MobileAdsModuleInterface;
-  _requestId: number;
-  _adUnitId: string;
-  _requestOptions: RequestOptions;
-  _loaded: boolean;
-  _isLoadCalled: boolean;
-  _onAdEventHandler: AdEventListener | null;
-  _nativeListener: EmitterSubscription;
-
-  constructor(
+  protected _type: 'app_open' | 'interstitial' | 'rewarded';
+  protected _googleMobileAds: MobileAdsModuleInterface;
+  protected _requestId: number;
+  protected _adUnitId: string;
+  protected _requestOptions: RequestOptions;
+  protected _loaded: boolean;
+  protected _isLoadCalled: boolean;
+  protected _adEventsListeners: Map<number, AdEventsListener<EventType>>;
+  protected _adEventListenersMap: Map<EventType, Map<number, AdEventListener<EventType>>>;
```

Comment:
Was wondering why `_adEventListenersMap` explicitly describes it's a Map and  `_adEventsListeners` not.
Not a big deal.

Reply:
I think it is because one of them is a map of listeners-by-type (map of maps) and the other is just a map of listeners?

Reply:
(the adEvents listeners is the mostly-backwards-compatible "just send me all the events" collection, while the non-plural adEvent listeners are the fully-split out "here's a listener for this specific type" ones)